src/compiler/parser.ts

```diff
@@ -613,19 +613,21 @@ namespace ts {
         let parseErrorBeforeNextFinishedNode = false;
 
         export function parseSourceFile(fileName: string, sourceText: string, languageVersion: ScriptTarget, syntaxCursor: IncrementalParser.SyntaxCursor, setParentNodes?: boolean, scriptKind?: ScriptKind): SourceFile {
+            const isDeclarationFile = isDeclarationFileName(fileName);

```

Comment:
Do we need to thread this through? Can we not just set the context flag in `createSourceFile`?